src/pydicom/pixels/utils.py

```diff
@@ -1738,6 +1738,188 @@ def reshape_pixel_array(ds: "Dataset", arr: "np.ndarray") -> "np.ndarray":
     return arr
 
 
+def set_pixel_data(
+    ds: "Dataset",
+    arr: "np.ndarray",
+    photometric_interpretation: str,
+    bits_stored: int,
+) -> None:
+    """Use an :class:`~numpy.ndarray` to set a dataset's *Pixel Data* and related
+    Image Pixel module elements.
+
+    .. versionadded:: 3.0
+
+    The following :dcm:`Image Pixel<part03/sect_C.7.6.3.3.html#table_C.7-11c>`
+    module elements values will be added, updated or removed as necessary:
+
+    * (0028,0002) *Samples per Pixel* using a value corresponding to
+      `photometric_interpretation`.
+    * (0028,0004) *Photometric Interpretation* from `photometric_interpretation`.
+    * (0028,0006) *Planar Configuration* will be added and set to ``0`` if
+      *Samples per Pixel* is > 1, otherwise it will be removed.
+    * (0028,0008) *Number of Frames* from the array :attr:`~numpy.ndarray.shape`,
+      however it will be removed if `arr` only contains a single frame.
+    * (0028,0010) *Rows* and (0028,0011) *Columns* from the array
+      :attr:`~numpy.ndarray.shape`.
+    * (0028,0100) *Bits Allocated* from the array :class:`~numpy.dtype`.
+    * (0028,0101) *Bits Stored* and (0028,0102) *High Bit* from `bits_stored`.
+    * (0028,0103) *Pixel Representation* from the array :class:`~numpy.dtype`.
+
+    In addition, the *Transfer Syntax UID* will be set to *Explicit VR Little
+    Endian* if it doesn't already exist or uses a compressed (encapsulated)
+    transfer syntax.
+
+    Parameters
+    ----------
+    ds : pydicom.dataset.Dataset
+        The dataset to modify.
+    arr : np.ndarray
+        An array with :class:`~numpy.dtype` uint8, uint16, int8 or int16. The
+        array must be shaped as one of the following:
+
+        * (rows, columns) for a single frame of grayscale data.
+        * (frames, rows, columns) for multi-frame grayscale data.
+        * (rows, columns, samples) for a single frame of multi-sample data
+          such as RGB.
+        * (frames, rows, columns, samples) for multi-frame, multi-sample data.
+    photometric_interpretation : str
+        The value to use for (0028,0004) *Photometric Interpretation*. Valid values
+        are ``"MONOCHROME1"``, ``"MONOCHROME2"``, ``"PALETTE COLOR"``, ``"RGB"``,
+        ``"YBR_FULL"``, ``"YBR_FULL_422"``.
+    bits_stored : int
+        The value to use for (0028,0101) *Bits Stored*. Must be no greater than
+        the number of bits used by the :attr:`~numpy.dtype.itemsize` of `arr`.
+    """
+    from pydicom.dataset import FileMetaDataset
+    from pydicom.pixels.common import PhotometricInterpretation as PI
+
+    if "FloatPixelData" in ds or "DoubleFloatPixelData" in ds:
+        raise AttributeError(
+            "The dataset has (7FE0,0008) 'Float Pixel Data' or (7FE0,0009) "
+            "'Double Float Pixel Data' elements which must first be deleted "
```

Comment:
I would expect that the dataset has a SOP class incompatible with setting `PixelData`, if one of these is present, so removing them would not help - at least if want to creater conformant DICOM.

Reply:
Hmm True. I'll change the exception message to say something like that

Reply:
Done